tests/wallet/test_singleton_store.py

```diff
@@ -0,0 +1,279 @@
+
+        # make record confirmed
+        confirmed_height = uint32(10)
+        timestamp = uint64(time.time())
+        await db.set_confirmed(record.coin_id(), confirmed_height, timestamp)
+        rec = await db.get_record_by_coin_id(record.coin_id())
+        assert isinstance(rec, SingletonCoinRecord)
+        assert rec.confirmed
+        assert rec.confirmed_at_height == confirmed_height
+
+        # get the next singleton record and add it to DB
+        next_record = get_next_singleton_record(record)
+        await db.add_singleton_record(next_record)
+
+        # check the new record exists and the old record has been set to spent
+        recs = await db.get_records_by_singleton_id(record.singleton_id)
+        assert len(recs) == 2
+        assert not recs[0].confirmed
+        assert recs[1].confirmed
+        assert recs[0].generation == 1
+        assert recs[1].spent_height == 0
+
+        # check get_latest_singleton with and without filtering confirmed
+        latest = await db.get_latest_singleton(singleton_id)
+        assert latest == recs[0]
+        latest = await db.get_latest_singleton(singleton_id, only_confirmed=True)
+        assert latest == recs[1]
+
+        # check get_unconfirmed_singletons
+        unconfirmed = await db.get_unconfirmed_singletons(singleton_id)
+        assert len(unconfirmed) == 1
+        assert unconfirmed[0] == recs[0]
+
+        # confirm the new record
+        await db.set_confirmed(next_record.coin_id(), uint32(20), uint64(time.time()))
+        recs = await db.get_records_by_singleton_id(record.singleton_id)
+        assert recs[0].confirmed
+        assert recs[1].spent_height == recs[0].confirmed_at_height == uint32(20)
+
+        # Delete the last record
+        await db.delete_record_by_coin_id(next_record.coin_id())
+        recs = await db.get_records_by_singleton_id(singleton_id)
+        assert len(recs) == 1
+        assert recs[0].coin_id() == record.coin_id()
+
+        # Delete all records by id
+        await db.delete_records_by_singleton_id(singleton_id)
+        recs = await db.get_records_by_singleton_id(singleton_id)
+        assert len(recs) == 0
+
+
+@pytest.mark.asyncio
+async def test_get_records_by_singleton_id() -> None:
+    async with DBConnection(1) as wrapper:
+        db = await WalletSingletonStore.create(wrapper)
+        one = uint32(1)
+
+        recs = await db.get_records_by_singleton_id(fake_singleton_id)
+        assert recs == []
+        record = get_new_singleton_record()
+        recs = await db.get_records_by_singleton_id(record.singleton_id)
+        assert len(recs) == 0
+        recs = await db.get_records_by_singleton_id(record.singleton_id, one, one, one)
+        assert len(recs) == 0
+        await db.add_singleton_record(record)
+        recs = await db.get_records_by_singleton_id(record.singleton_id)
+        assert len(recs) == 1
+        recs = await db.get_records_by_singleton_id(record.singleton_id, one, one, one)
```

Comment:
this function has 3 optional constraints.
```
        min_generation: Optional[uint32] = None,
        max_generation: Optional[uint32] = None,
        num_results: Optional[uint32] = None,
```

as far as I can tell, there are no tests ensuring these work as intended.